### ISSUE TITLE
Compass as marker?

### ISSUE THREAD

**Issue by dennisguse** [OpenTracksApp/OSMDashboard#60]
I really love the compass that we have!
However, the current implementation is not really useful for the PiP mode (looks kind of not really cool and that's why it is hidden; also due to the lack of space).

I have no clue how mapsforge works, but could not we mark the current position (the one that is currently marked with a red dot) similar as we put the marker(s) on the map?
I mean the marker pins are zoom invariant and do not seem to incur rendering performance issues, right?
Long story short: can render the compass arrow as a "marker" and just rotate it?
And when we get a new position, we either move or delete/add the compass arrow again.

PS: I was just thinking aloud.

**Comment by pstorch** [comment]
Nice idea

**Comment by pstorch** [comment]
@dennisguse happy testing :wink:

**Comment by dennisguse** [comment]
Almost there: arrows is at the correct position and shows the traveling direction (using the last two trackpoints I guess)
But I meant to actually also use the compass orientation for this arrow (in terms of replacing the compassView).

**Comment by pstorch** [comment]
Oh, then I misunderstood.
But actually I like it that way. 😁
To have both the travel direction and the view orientation.
Let me think what we can do.

**Comment by pstorch** [comment]
@dennisguse ok, second try :wink:

**Comment by dennisguse** [comment]
And it is back ;)
Just did some testing and it works great.
Only thing  the compass orientation is only updated, if you are moving (right?).
So, if you are standing still (and OpenTracks does add new TrackPoints), then the compass does not get updated.

I just tested it, holding my phone out of the window ;)
And if was only updating now and then, but not when I rotated the device.

PS: i confirmed that my compass was working with "Trail Sense" - sometimes the compass is really off on my two devices ;)

**Comment by pstorch** [comment]
Hmm, the update is working on my device without moving. Maybe the update (redraw) works differently on different devices. Let me check. I'm also working on #61 that interferes here. So I'll work on both.

**Comment by dennisguse** [comment]
No hurry - I already love the arrow (just by direction without the compass).

**Comment by pstorch** [comment]
@dennisguse because of the update problem: you have set the Arrow mode in the menu to "point to north", right?

**Comment by dennisguse** [comment]
Yes, it was "point to north" - didn't k ow there was a setting ;)

**Comment by pstorch** [comment]
Ok, that was my silent hope for the update problem. Because the direction arrow only updates on new Track point, but the north arrow should update live.
